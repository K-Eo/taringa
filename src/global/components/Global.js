import React from 'react'

import Ad from '../../ads/components/Ad'
import Feed from '../../feed/components/Feed'
import Filterable from '../../filters/components/Filterable'

const filters = {
  hot: {
    displayName: 'Destacados',
    exact: true,
    id: 'hot',
    pathname: '/global',
    url: '/feed/global',
  },
  recents: {
    displayName: 'Recientes',
    exact: false,
    id: 'recents',
    pathname: '/global/recents',
    url: '/feed/global?sort=recent',
  },
  tops: {
    displayName: 'Tops',
    exact: false,
    id: 'tops',
    pathname: '/global/tops',
    url: '/feed/global/tops/week',
  },
}

const Global = ({ match }) => {
  const filter = match.params.filter || 'hot'
  const url = filters[filter].url

  return (
    <div className="row">
      <Filterable filters={filters} />

      <div className="col-8">
        <Feed feedId="global" filter={filter} url={url} />
      </div>

      <div className="col-4">
        <Ad />
      </div>
    </div>
  )
}

Global.defaultProps = {
  match: { params: {} },
}

export default Global
