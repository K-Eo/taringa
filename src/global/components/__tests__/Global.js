import React from 'react'

import Global from '../Global'

describe('Global', () => {
  it('exists', () => {
    expect(Global).toBeDefined()
  })

  it('renders', () => {
    const wrapper = shallow(<Global />)

    expect(wrapper).toMatchSnapshot()
  })
})
