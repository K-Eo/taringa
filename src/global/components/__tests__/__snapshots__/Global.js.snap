// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global renders 1`] = `
<div
  className="row"
>
  <Connect(Filterable)
    filters={
      Object {
        "hot": Object {
          "displayName": "Destacados",
          "exact": true,
          "id": "hot",
          "pathname": "/global",
          "url": "/feed/global",
        },
        "recents": Object {
          "displayName": "Recientes",
          "exact": false,
          "id": "recents",
          "pathname": "/global/recents",
          "url": "/feed/global?sort=recent",
        },
        "tops": Object {
          "displayName": "Tops",
          "exact": false,
          "id": "tops",
          "pathname": "/global/tops",
          "url": "/feed/global/tops/week",
        },
      }
    }
  />
  <div
    className="col-12 col-lg-8"
  >
    <Connect(FilterableInfiniteScroll)
      feedId="global"
      filter="hot"
      url="/feed/global"
    />
  </div>
  <div
    className="col-4 order-first order-lg-last d-none d-lg-block"
  >
    <Ad />
  </div>
</div>
`;
